### PR TITLE
Fix some issues with error handling on the check your answers page

### PR DESCRIPTION
**Pull request by cbaines**
Because Rails, there's inconsistencies with how loading things works
in different environments (autoloading vs eagerloading). I think this
was leading to problems with loading the errors.rb file, at least when
running the tests.

To address this, this commit moves the things in that file in to a
module named like Rails would expect from the filename. This should
mean that Rails finds the right file when attempting to autoload
things.

This commit also changes a few other things that seemed to error, and
adds a test for the case where there's insufficient session data when
the post request is made.

### CODE REVIEW

File: app/lib/errors.rb [hunk 1]
```diff
@@ -1,21 +1,23 @@
-class EmailTooLongError < StandardError
-end
+module Errors
```
Comment (Krenair) [inline app/lib/errors.rb:1]: there's various references to things in this module that are outside check_your_answers_controller - don't we now need to go and update them all?

**Comment by Krenair** [comment]
looks good other than the inline comment

**Comment by Krenair** [comment]
also log_error?

**Review by Krenair — APPROVED** [review]

File: app/lib/errors.rb (hunk 1 above)
Comment (cbaines) [reply app/lib/errors.rb:1]: Yep, I've done that now :)

**Comment by cbaines** [comment]
> also log_error?

Yeah, don't know why I didn't look at that... it's fixed now.

**Pull request merged**